docs/data-sources/cluster.md

```diff
@@ -16,22 +16,110 @@ Data source for a Redpanda Cloud cluster
 
 - `id` (String) The ID of the cluster
 
+### Optional
+
+- `aws_private_link` (Attributes) (see [below for nested schema](#nestedatt--aws_private_link))
+- `gcp_private_service_connect` (Attributes) (see [below for nested schema](#nestedatt--gcp_private_service_connect))
+
 ### Read-Only
 
 - `allow_deletion` (Boolean) allows deletion of the cluster. defaults to true. Not recommended for production use
-- `cloud_provider` (String) Must be one of aws or gcp
+- `cloud_provider` (String) Must be one of aws, azure or gcp
 - `cluster_api_url` (String) The URL of the cluster API
 - `cluster_type` (String) Type of the cluster
 - `connection_type` (String) Connection type of the cluster
+- `http_proxy` (Attributes) (see [below for nested schema](#nestedatt--http_proxy))
+- `kafka_api` (Attributes) (see [below for nested schema](#nestedatt--kafka_api))
```

Comment:
The nested schema below does not indicate what are these field for.

By looking at the code I can see that these and `schema_registry` are for enabling mtls and not for setting anything on the listener. One would think with this naming that we have schema registry support, or a way to setup the listeners configuration, but no.

What do you think of having a top-level property: `mtls` with the listeners below them?
````
mtls
  - kafka
    - enabled
    - ca_certificates_pem
    - ...
  - http_proxy
    - ... 
  - schema_registry
```

Reply:
I'm following the API pattern pretty strictly here because... I mean yeah I completely agree 😂  

My worry is that we diverge, they change something so that there is config alongside the MTLS and we have a refactor.